packages/noir-ethereum-api/test/ethereum/recordingClient.test.ts

```diff
@@ -0,0 +1,33 @@
+import { describe, expect, it } from 'vitest';
+import { createDefaultClient } from '../../src/ethereum/client.js';
+import { AwaitedCallResultEntry, createRecordingClient } from '../../src/ethereum/createRecordingClient.js';
+import { GetBlockReturnType, GetProofReturnType } from 'viem';
+
+describe('recordingClient', () => {
+  it('record JSON-RPC API calls', async () => {
+    const client = createRecordingClient(createDefaultClient());
```

Comment:
After additional consideration I would mock defaultClient(), but let's do in the next PR with MockClient